Arriba/Arriba.Client/Model/SecureDatabase.cs

```diff
@@ -51,6 +53,142 @@ public SecurityPermissions Security(string tableName)
             }
         }
 
+        public override T Query<T>(IQuery<T> query)
+        {
+            throw new ArribaException("Use Query overload which takes isCurrentUserIn on SecureDatabase.");
+        }
+
+        public T Query<T>(IQuery<T> query, Func<SecurityIdentity, bool> isCurrentUserIn)
```

Comment:
What is the purpose of this API if SecureQuery is also public?  Why not just force them into SecureQuery?